From 34ff4be2aac6d7da8d61b0c5538309996ac9d416 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 12 Feb 2014 13:07:12 +0800
Subject: [PATCH 56/68] ENGR00294739 coreimage, screenrecord sometimes fail
 when shift different video

The recorder exit because can't request encoder buffer. In this case,
sample table is not written to the mp4 file, so can't play.
Make sure the sample table is written can fix it.

Why can't request encoder buffer?
Because during shift to other clips, the media scan is working.
In the sd card, a special mkv clip will cause mediaserver process deliberately
aborted.

The abort check is as below in MatroskaExtractor.cpp
"CHECK_GE(avccSize, 5u);"

In the sdcard, there's a mkv clip will cause

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 cmds/screenrecord/screenrecord.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/cmds/screenrecord/screenrecord.cpp b/cmds/screenrecord/screenrecord.cpp
index 49999b5..e2ea504 100644
--- a/cmds/screenrecord/screenrecord.cpp
+++ b/cmds/screenrecord/screenrecord.cpp
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ */
+
 #define LOG_TAG "ScreenRecord"
 //#define LOG_NDEBUG 0
 #include <utils/Log.h>
@@ -501,9 +505,11 @@ static status_t recordScreen(const char* fileName) {
     // Main encoder loop.
     err = runEncoder(encoder, muxer);
     if (err != NO_ERROR) {
+        // If err occurs, make sure the mp4 file is ended correctly,
+        // In other words, sample table is written to the end of file.
+        encoder->stop();
+        muxer->stop();
         encoder->release();
-        encoder.clear();
-
         return err;
     }
 
-- 
1.8.0

